bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/TextLayout.java

```diff
@@ -2121,7 +2121,7 @@ public FontMetrics getLineMetrics (int lineIndex) {
 	lptm.tmHeight = DPIUtil.autoScaleUp(getDevice(), ascentInPoints + descentInPoints);
 	lptm.tmInternalLeading = DPIUtil.autoScaleUp(getDevice(), leadingInPoints);
 	lptm.tmAveCharWidth = 0;
-	return FontMetrics.win32_new(lptm);
+	return FontMetrics.win32_new(lptm, DPIUtil.getNativeDeviceZoom());
```

Comment:
This looks wrong to me: why is it valid to use the zoom from non-multi-monitor-aware `DPIUtil`? Won't this fail in the multi-monitor use case?